addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/handler/solarviewHandler.java

```diff
@@ -0,0 +1,417 @@
+     */
+    private State getChannelValue(String id) {
+        logger.trace("getChannelValue() called for channel {}.", id);
+        int itemIndex;
+        /**
+         * Sample input from Solarview server:
+         * {00,07,06,2017,12,26,0008.2,00169,002467,00024890,00121,000,000.0,000,000.0,000,000.0,000,000.0,00},<
+         *
+         * Explanation:
+         * {WR,Day,Month,Year,Hour,Minute,KDY,KMT,KYR,KT0,PAC,UDC,IDC,UDCB,IDCB,UDCC,IDCC,UL1,IL1,TKK}
+         */
+        switch (id) {
+            case CHANNEL_KDY:
+                itemIndex = 6;
+                break;
+            case CHANNEL_KMT:
+                itemIndex = 7;
+                break;
+            case CHANNEL_KYR:
+                itemIndex = 8;
+                break;
+            case CHANNEL_KT0:
+                itemIndex = 9;
+                break;
+            case CHANNEL_PAC:
+                itemIndex = 10;
+                break;
+            case CHANNEL_UDC:
+                itemIndex = 11;
+                break;
+            case CHANNEL_IDC:
+                itemIndex = 12;
+                break;
+            case CHANNEL_UDCB:
+                itemIndex = 13;
+                break;
+            case CHANNEL_IDCB:
+                itemIndex = 14;
+                break;
+            case CHANNEL_UDCC:
+                itemIndex = 15;
+                break;
+            case CHANNEL_IDCC:
+                itemIndex = 16;
+                break;
+            case CHANNEL_UL1:
+                itemIndex = 17;
+                break;
+            case CHANNEL_IL1:
+                itemIndex = 18;
+                break;
+            case CHANNEL_TKK:
+                itemIndex = 19;
+                break;
+            default:
+                logger.error("getChannelValue() called for unknown channel: {}", id);
+                return UnDefType.UNDEF;
+        }
+        if (energyInformation != null) {
+            String value = getNthValue(energyInformation, itemIndex);
+            DecimalType val;
+            if (value != null) {
+
+                logger.trace("getChannelValue() returns {}.", value);
+
+                try {
+                    val = new DecimalType(value);
+                } catch (NumberFormatException nx) {
+                    val = null;
+                }
+                return (val);
+            }
+        }
+
+        logger.trace("getChannelValue() returns UNDEF.");
+
+        return UnDefType.UNDEF;
+    }
+
+    /**
+     * Split the passed strings (separated by the delimiter SOLARVIEW_RESPONSE_ITEM_DELIM)
+     * into parts and return the nth item.
+     *
+     * @param string to be splitted
+     * @return the value of the n-th part, or, in case of failure, UNDEF.
+     */
+    private String getNthValue(String data, int nth) {
+        logger.trace("getNthValue() called.");
+        String[] part = (StringUtils.split(data, SOLARVIEW_RESPONSE_ITEM_DELIM));
```

Comment:
You're completely right. This was a waste of resources. Thks.

Fixed by string separation during/after communication.